reconcile/test/test_openshift_resources_base.py

```diff
@@ -482,3 +482,21 @@ def test_cluster_params():
 
     with pytest.raises(RuntimeError):
         orb.run(dry_run=False, cluster_name=["cluster-1", "cluster-2"])
+
+
+@pytest.mark.parametrize(
+    "test_keyname, exception_expected",
+    [
+        (" leading_space", orb.SecretKeyFormatError),
+        (" space_padding ", orb.SecretKeyFormatError),
+        ("trailing_space ", orb.SecretKeyFormatError),
+        ("no space issues", None),  # but should there be...
+        ("no_spacing", None),
+    ],
+)
+def test_secret_keys(test_keyname, exception_expected):
```

Comment:
I suggest to split up the tests so that they can verify one behavior at a time. In this case, I suggest you create a separate test that expects the error and another one without error.

Reply:
I've decided not to do that for two reasons. First, I want to avoid future comments similar to https://github.com/app-sre/qontract-reconcile/pull/3967#discussion_r1431064114. Second, I think the test table approach here is a best fit because the tests all follow the same structure and flow.

Reply:
I highly suggest reading https://testing.googleblog.com/2014/04/testing-on-toilet-test-behaviors-not.html.

You can still utilize parametrize tests for different inputs, just isolate the two different behavior of the same function in different tests :) 

That said, I leave this up to you if you still want to keep the same test.